### PR TITLE
Add repositories search box

### PR DESCRIPTION
**Pull request by aeisenberg**
A simple, webview-only search box for filtering repositories from the remote queries results view.

- [x] ~TODO: Need to handle dark mode.~ 

Do we want any animations for showing/hiding repos?

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+interface RepositoriesSearchProps {
+  filterValue: string;
+  setFilterValue: (value: string) => void;
+}
+
+const RepositoriesSearch = ({ filterValue, setFilterValue }: RepositoriesSearchProps) => {
+  return <>
+    <TextInput
+      block
+      sx={{
```
Comment (charisk) [inline extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:15]: This doesn't look like it's needed. We what it looks like without it and it looks fine in both modes to us.

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        backgroundColor: 'var(--vscode-editor-background);',
+        color: 'var(--vscode-editor-foreground);'
+      }}
+      className="vscode-codeql__repositories-search"
```
Comment (charisk) [inline extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:19]: This class is added to add some space to the top. I've generally used `<VerticalSpace />` for that e.g. `<VerticalSpace size={2} />`

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { SearchIcon } from '@primer/octicons-react';
+
+interface RepositoriesSearchProps {
+  filterValue: string;
+  setFilterValue: (value: string) => void;
+}
+
+const RepositoriesSearch = ({ filterValue, setFilterValue }: RepositoriesSearchProps) => {
+  return <>
+    <TextInput
```
Comment (charisk) [inline extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:13]: In a large screen this looks a bit silly. Shall we set max-width? e.g. `max-width: 30em`.

image

**Review by charisk — COMMENTED** [review]
Looking good, just some minor comments.

Reviewed with @robertbrignull @shati-patel

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 3 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:13]: Should we have a max-width for the entire results view and just let the search box follow that?

The wireframes have the search box exactly as wide as the NWO results.

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:19]: Sure, I can change that. Just wondering what the benefit of using extra dom for adding space instead of using css.

**Comment by aeisenberg** [comment]
@charisk what do you think of the latest commit? It adds a max-width to the entire page and all sub-elements respect that value. Now the search bar is always _exactly_* as wide as the results below it.

* Actually, I need to change from % to fixed difference since on narrow screens, the values are not correct. Let me fix that.

**Comment by aeisenberg** [comment]
done

File: extensions/ql-vscode/src/remote-queries/view/remoteQueries.css [hunk 4]
```diff
@@ -41,7 +45,6 @@
 }
 
 .vscode-codeql__repositories-search {
```
Comment (charisk) [inline extensions/ql-vscode/src/remote-queries/view/remoteQueries.css:47]: This can be deleted now.

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 2 above)
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:20]: IMO it's much easier to follow/reason with rather than poking with margins and padding (and dealing with conflicting margins, different context/layout etc.)

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:15]: Hmm yeah I have no idea what it should look like in red 😂 we just checked light and dark mode. Ideally we'd be setting colors through the Primer theme but that's something to look into separately!

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx [hunk 5]
```diff
@@ -14,7 +14,8 @@ const RepositoriesSearch = ({ filterValue, setFilterValue }: RepositoriesSearchP
       block
       sx={{
         backgroundColor: 'var(--vscode-editor-background);',
-        color: 'var(--vscode-editor-foreground);'
+        color: 'var(--vscode-editor-foreground);',
+        width: 'calc(100% - 14px)',
```
Comment (charisk) [inline extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:18]: Why is this needed? If I turn it off things look the same in my view. Is it trying to add a padding of some sort?

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 3 above)
Comment (charisk) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:13]: Good call!

**Review by charisk — APPROVED** [review]

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 5 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:18]: Yes. The width of the search bar needs to be exactly the same as the width of the analysis results section. I was trying to find a way to get this to work using padding or margin, but because we are using primer, we can't attach css to the elements that would be affected by padding or margin elements. This is the only way I could figure out how to get it working.

File: extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/remote-queries/view/RepositoriesSearch.tsx:15]: Handling all kinds of themes will be very, very tricky. But, I think we should at least make a best effort for the common themes (though, I don't know if red is very common).
